packages/drivers/driver-base/package.json

```diff
@@ -14,12 +14,20 @@
 	"type": "module",
 	"exports": {
 		".": {
+			"trimmedAPI": {
+				"import": {
+					"types": "./lib/public.d.ts"
+				},
+				"require": {
+					"types": "./dist/public.d.ts"
+				}
+			},
```

Comment:
This doesn't seem quite right.

Reply:
I see them in the original change, but I'm not sure what they're for.

Reply:
`trimmedAPI` is an import condition. If someone wanted to opt-in to trimmed APIs while building they could specify "trimmedAPI" and then the condition would be satisfied and those options would be chosen. This is a way to enforce respect for the trimming without forcing upon all consumers.
If this is going to be done, it should be consistent throughout.

Reply:
Nicer way to write it is
```
			"trimmedAPI": {
				"types": {
					"import": "./lib/public.d.ts",
					"require": "./dist/public.d.ts"
				}
			},
```

Reply:
> If someone wanted to opt-in to trimmed APIs while building they could specify "trimmedAPI" and then the condition would be satisfied and those options would be chosen.

Unless we have customers doing this or expect Loop to do it, then I think we should just point the types to index.d.ts and call it a day.

Reply:
Although we don't expect any customers to be using it, it may be worth keeping it just to stay consistent with RC3. That way in the off chance that someone was actually using it (although it seems very unlikely), they won't suddenly break in RC4

Reply:
But we don't do it consistently in RC3. Some packages have it while others don't. Why? I don't see how anyone could be using this since we haven't ever documented it, and it was apparently added in an RC3 patch.

Reply:
Turns out when I remove them, I get the following build errors:
```
Failed Tasks:
[ 290/1208] x @fluidframework/routerlicious-urlresolver: flub generate entrypoints --outFileAlpha legacy --outDir ./lib --node10TypeCompat - 2.723s
[ 312/1208] x @fluidframework/routerlicious-urlresolver: flub generate entrypoints --outFileAlpha legacy --outDir ./dist - 2.820s
[ 381/1208] x @fluidframework/driver-base: flub generate entrypoints --outFileAlpha legacy --outDir ./lib --node10TypeCompat - 2.673s
[ 384/1208] x @fluidframework/driver-base: flub generate entrypoints --outFileAlpha legacy --outDir ./dist - 2.445s
[ 426/1208] x @fluidframework/replay-driver: flub generate entrypoints --outFileAlpha legacy --outDir ./lib --node10TypeCompat - 2.945s
[ 427/1208] x @fluidframework/replay-driver: flub generate entrypoints --outFileAlpha legacy --outDir ./dist - 2.784s
[ 446/1208] x @fluidframework/tool-utils: flub generate entrypoints --outFileAlpha legacy --outDir ./lib --node10TypeCompat - 2.710s
[ 451/1208] x @fluidframework/tool-utils: flub generate entrypoints --outFileAlpha legacy --outDir ./dist - 2.611s
[ 480/1208] x @fluidframework/file-driver: flub generate entrypoints --outFileAlpha legacy --outDir ./dist - 1.807s
```

What's interesting is that some packages where I removed `trimmedAPI` (i.e. fluid-static) did not get this build error despite having the same `api-extractor:commonjs` and `api-extractor:esnext` commands.

Reply:
Very strange... I see that RC4 is on an earlier build-tools - I wonder if it's missing some changes in generate entrypoints. I'll look a bit more.

Reply:
The error from the cell package when you remove the trimmedApi condition:

```
[127/133] x @fluidframework/cell: flub generate entrypoints --outFileAlpha legacy --outDir ./lib --node10TypeCompat - 1.366s
@fluidframework/cell: error during command 'flub generate entrypoints --outFileAlpha legacy --outDir ./dist' (exit code 1)
@fluidframework/cell: Error: There are no package exports matching requested output entrypoints:
@fluidframework/cell:           dist/legacy.d.ts
@fluidframework/cell:           dist/beta.d.ts
@fluidframework/cell:           dist/public.d.ts
```

I can also work around those failures by adding a ./public entrypoint that points to public.d.ts. So I think the evidence points to `generate entrypoints` needing something in the exports that points to all the entrypoints. Let's talk to Jason at standup.

The advantage of the current approach is that the APIs are _only_ accessible when using this custom condition, so they're much better hidden than if we added a ./public entrypoint or something like it. I'm coming around to this approach - I just want to document it somewhere (for ourselves, so we don't ask these same questions in 2 months).

Reply:
Upgrading to build-tools 0.39.0 is also an option because I can't repro the failures in that version, but that carries with it other changes. We could also consider porting the generate entrypoints changes to 0.38 and release a patch. But all of that is more complex than this workaround.

Reply:
The trimmedApi entries should NOT be needed in the RC5 version of this change, though.